Detectors/ZDC/calib/src/BaselineCalibSpec.cxx

```diff
@@ -0,0 +1,147 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file   BaselineCalibSpec.cxx
+/// @brief  ZDC baseline calibration
+/// @author pietro.cortese@cern.ch
+
+#include <iostream>
+#include <vector>
+#include <string>
+#include <gsl/span>
+#include "CCDB/BasicCCDBManager.h"
+#include "CCDB/CCDBTimeStampUtils.h"
+#include "CCDB/CcdbApi.h"
+#include "Framework/Logger.h"
+#include "Framework/ControlService.h"
+#include "Framework/ConfigParamRegistry.h"
+#include "Framework/CCDBParamSpec.h"
+#include "Framework/DataRefUtils.h"
+#include "DetectorsCommonDataFormats/DetID.h"
+#include "ZDCBase/ModuleConfig.h"
+#include "CommonUtils/NameConf.h"
+#include "CommonUtils/MemFileHelper.h"
+#include "CCDB/BasicCCDBManager.h"
+#include "CCDB/CCDBTimeStampUtils.h"
+#include "ZDCCalib/BaselineCalibSpec.h"
+#include "ZDCCalib/BaselineCalibData.h"
+
+using namespace o2::framework;
+
+namespace o2
+{
+namespace zdc
+{
+
+BaselineCalibSpec::BaselineCalibSpec()
+{
+  mTimer.Stop();
+  mTimer.Reset();
+}
+
+BaselineCalibSpec::BaselineCalibSpec(const int verbosity) : mVerbosity(verbosity)
+{
+  mTimer.Stop();
+  mTimer.Reset();
+}
+
+void BaselineCalibSpec::init(o2::framework::InitContext& ic)
+{
+  mVerbosity = ic.options().get<int>("verbosity-level");
+  mWorker.setVerbosity(mVerbosity);
+}
+
+void BaselineCalibSpec::updateTimeDependentParams(ProcessingContext& pc)
```

Comment:
I don't see `finaliseCCDB` declared or implemented, w/o it the CCDB objects will never end up in your code.

Reply:
I'm implementing this but it seems I'm missing something. The program looks for objects that are valid today and not at the timestamp of the simulation. 
[o2-zdc-digits-reco.out.gz](https://github.com/AliceO2Group/AliceO2/files/8975255/o2-zdc-digits-reco.out.gz)

Reply:
The fetcher looks for the timestamp mentioned in the OrbitReset object + offset for the elapsed orbit. 
Since you are using local ccdb, I cannot tell what you have there
```
[239655:internal-dpl-ccdb-backend]: [11:50:28][INFO] ccdb reads http://localhost:8080/CTP/Calib/OrbitReset/0/ac738440-f38e-11ec-a902-7f0000015566 for 1647007420614 (retrieve, agent_id: fis01.disit.uniupo.it-1656064227-dFPJ6C), 
[239655:internal-dpl-ccdb-backend]: [11:50:28][INFO] ccdb reads http://localhost:8080/ZDC/Calib/BaselineCalib/1550600800000/25a9d700-f39f-11ec-a902-7f0000015566 for 1656055464565 (retrieve, agent_id: fis01.disit.uniupo.it-1656064227-dFPJ6C), 
[239655:internal-dpl-ccdb-backend]: [11:50:28][INFO] ccdb reads http://localhost:8080/ZDC/Config/Module/1550600800000/f9597310-f38d-11ec-a902-7f0000015566 for 1656055464565 (retrieve, agent_id: fis01.disit.uniupo.it-1656064227-dFPJ6C), 
```

Reply:
Thanks for the info. I was able to make it working by some trial and error.
I have a last question. Where do I pick up the correct time for a simulation in order to write a proper OrbitReset object? It is something related to simulation time but I don't understand where it is stored. Thanks